File: server/src/Models/roleModel.js
import mongoose from "mongoose";

const roleSchema = new mongoose.Schema({
  role: {
    type: String,
    required: true,
    unique: true,
  },
  sector: {
    type: String,
    required: true,
  },
  baseSalary: {
    type: Number,
    required: true,
    min: 0,
  },
  additives: [String],
  isField: {
    type: Boolean,
    default: false,
    required: true,
  },
});

roleSchema.statics.isUnique = async function (role) {
  try {
    if (!role) throw new Error("Dados inválidos");

    const roleInDatabase = await this.findOne({ role });

    if (roleInDatabase) return false;
    return true;
  } catch (err) {
    console.log("Erro ao verificar dados", err.message);
    return false;
  }
};

export default mongoose.model("Role", roleSchema);

File: server/src/models/roleModel.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type Role struct {
	ID         primitive.ObjectID `bson:"_id,omitempty" json:"id" validate:"required"`
	Role       string             `bson:"role" json:"role" validate:"required"`
	Sector     string             `bson:"sector" json:"sector" validate:"required"`
	BaseSalary float64            `bson:"baseSalary" json:"baseSalary" validate:"required"`
	Additives  []string           `bson:"additives,omitempty" json:"additives,omitempty"`
	IsField    bool               `bson:"isField" json:"isField" validate:"required"`
}
